engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/desugar/ComplexType.scala

```diff
@@ -125,6 +125,7 @@ case object ComplexType extends IRPass {
           lastAnnotations = Seq()
           res
         case _ =>
+          lastAnnotations = Seq()
```

Comment:
Unexpected place of fix - I was searching for the problem in `GenericAnnotations`...

Reply:
The annotation was lost here in the `ComplexType` compiler phase, and later `GenericAnnotations` was processing IR without the annotation node.